pkg/opentelemetry/controller.go

```diff
@@ -98,26 +104,20 @@ func NewController() (*Controller, error) {
 		resource.WithAttributes(
 			semconv.ServiceNameKey.String(serviceName),
 			semconv.TelemetrySDKLanguageGo,
+			semconv.TelemetryAutoVersionKey.String(releaseVersion),
 		),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	log.Logger.V(0).Info("Establishing connection to OpenTelemetry collector ...")
-	timeoutContext, cancel := context.WithTimeout(ctx, time.Second*10)
-	defer cancel()
-	conn, err := grpc.DialContext(timeoutContext, endpoint, grpc.WithInsecure(), grpc.WithBlock())
-	if err != nil {
-		log.Logger.Error(err, "unable to connect to OpenTelemetry collector", "addr", endpoint)
-		return nil, err
-	}
-
-	traceExporter, err := otlptracegrpc.New(ctx,
-		otlptracegrpc.WithGRPCConn(conn),
+	log.Logger.V(0).Info("Establishing connection to OTLP receiver ...")
+	otlpTraceClient := otlptracegrpc.NewClient(
```

Comment:
:tada: This constructor does all the heavy lifting of looking up and parsing environment variables for the exporter config. 🎉